language/move-lang/src/parser/syntax.rs

```diff
@@ -791,7 +731,7 @@ fn parse_term(tokens: &mut Lexer) -> Result<Exp, Error> {
 
         Tok::IdentifierValue => parse_name_exp(tokens)?,
 
-        Tok::AddressValue => {
+        Tok::NumValue => {
```

Comment:
Ah, clever: distinguishing the typed number token avoids the ambiguity here! I like it.